File: C_Code/01.1.1_Blink/Blink3.c

#include <pigpio.h>
#include <stdio.h>
#include <unistd.h>
#include <signal.h>

static volatile int keepRunning = 1;

void intHandler(int dummy) {
    keepRunning = 0; 
}


int main(int argc, char *argv[])
{
   int GPIO=17;
   int level;

   if (gpioInitialise() < 0) return 1;

   //level = gpioRead(GPIO);
	
	gpioSetMode(GPIO, PI_OUTPUT);       // Set GPIO17 as output
	printf("Using GPIO-pin %d\n",GPIO);	// Output information on terminal
	signal(SIGINT, intHandler);			// upon ^C the signal function is called to terminate the process;
										// intHandler will change the variable keepRunning form 1 to 0
										// now the infinite while loop will end, and main will stop gracefully
	while(keepRunning){
		gpioSetMode(GPIO, PI_OUTPUT);   // Set GPIO17 as output
		gpioWrite(GPIO,1);              // Make GPIO output HIGH level
		printf("led turned on\n");		// Output information on terminal
		sleep(1);						// Wait for 1 second
		gpioWrite(GPIO,0);              // Make GPIO output LOW level
		printf("led turned off\n");		// Output information on terminal
		sleep(5);						// Wait for 5 seconds
	}
	gpioTerminate();
	printf("\nBye\n");
	return 0;
}

File: Python_Code/01.1.1_Blink/Blink3.py
import pigpio
import time


ledPin = 17             # define ledPin to BCM GPIO17
pi1 = pigpio.pi()       # pi1 accesses the local Pi's GPIO


def setup():
    print ('using pin%d' %ledPin)


def loop():
    while True:
        pi1.write(ledPin, 0)         # set local Pi's GPIO 4 low
        print ('led turned off')     # print information on terminal
        time.sleep(3)                # Wait for 3 second
        pi1.write(ledPin, 1)         # make ledPin output LOW level to turn off led
        print ('led turned on')
        time.sleep(1)                # Wait for 1 second


def destroy():
    pi1.stop()                       # Release all GPIO


if __name__ == '__main__':           # Program entrance
    print ('Program is starting ... \n')
    setup()
    try:
        loop()
    except KeyboardInterrupt:        # Press ctrl-c to end the program.
        print("\n program stopped by Keyboard Interrupt")
        
    finally:
        destroy()